### PR TITLE
feat: support undoing all changes

### PR DESCRIPTION
**Pull request by maxwfreu**

### CODE REVIEW

File: components/editor/publish-button.tsx [hunk 1]
```diff
@@ -59,6 +62,13 @@ const PublishButton = ({
     log.debug(`Published Adventure: ${adventureId}`);
   };
 
+  const undoChanges = async () => {
+    setIsUndoLoading(true);
+    await resetDevAgent(adventureId as string);
+    window.location.reload();
```
Comment (maxwfreu) [inline components/editor/publish-button.tsx:68]: path invalidation isn't working properly here for some reason.. so for now a hard reload will do the trick

File: app/actions/adventure.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+"use server";
+import prisma from "@/lib/db";
+import { TopLevelSpecialCases } from "@/lib/editor/DEPRECATED-editor-options";
+import { auth } from "@clerk/nextjs";
+
+export const resetDevAgent = async (id: string) => {
+  const { userId } = auth();
+  if (!userId) throw new Error("No user");
+
+  const adventure = await prisma.adventure.findUnique({
```
Comment (eob) [inline app/actions/adventure.ts:10]: Not important but you can use `getAdventure`

**Review by eob — APPROVED** [review]
LGTM!

File: app/actions/adventure.ts (hunk 2 above)
Comment (maxwfreu) [reply app/actions/adventure.ts:10]: Yeah but that includes some stuff I don't really care about, like `devAgent` and `NarrativeSnippet`

**Pull request merged**